### PR TITLE
RISC-V header cleanups

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

File: include/zephyr/arch/riscv/irq.h [hunk 1]
```diff
@@ -32,6 +32,12 @@ extern "C" {
 /** Environment Call from M-mode */
 #define RISCV_EXC_ECALLM 11
 
+#ifdef CONFIG_64BIT
+#define RISCV_MCAUSE_IRQ_BIT          (1 << 63)
+#else
+#define RISCV_MCAUSE_IRQ_BIT          (1 << 31)
+#endif
```
Comment (carlocaione) [inline include/zephyr/arch/riscv/irq.h:46]: yeeeeaahh, we both now at least one platform where this is not true ...

**Review by carlocaione — DISMISSED** [review]

**Review by npitre — DISMISSED** [review]

**Review by carlocaione — DISMISSED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]
LGTM

**Comment by ycsin** [comment]
I'm doing an update to v3.7 and my application doesn't build, partially due to the changes in this PR. 

Given that the `soc.h` is exposed to the application/driver/subsys, can we provide the old macros removed/renamed in this PR and use the deprecation process?

cc @carlocaione @gmarull

**Comment by gmarull** [comment]
> I'm doing an update to v3.7 and my application doesn't build, partially due to the changes in this PR.
> 
> Given that the `soc.h` is exposed to the application/driver/subsys, can we provide the old macros removed/renamed in this PR and use the deprecation process?
> 
> cc @carlocaione @gmarull

I don't think the content of soc.h can be considered "stable", soc.h has been largely removed as it introduces all sorts of problems.

**Comment by ycsin** [comment]
> I don't think the content of soc.h can be considered "stable"

Sure, but these common macros have been there for years and exposed to ..everything, it isn't fun to find the applications fail to compile in a sea of compilation error and have to dig through the versions/commits/PRs to find where they are.

**Comment by gmarull** [comment]
> > I don't think the content of soc.h can be considered "stable"
> 
> Sure, but these common macros have been there for years and exposed to ..everything, it isn't fun to find the applications fail to compile in a sea of compilation error and have to dig through the versions/commits/PRs to find where they are.

Applications should not rely on soc.h ;-)

**Comment by ycsin** [comment]
> Applications should not rely on soc.h ;-)

I meant the whole thing really. Application doesn't compile if SOC/boards/drivers/subsys/lib doesn't compile ¯\\_(ツ)_/¯ The changes here can break OOT stuff that has something to do with interrupt/exception, i.e. PLIC driver

It becomes much more complicated when the OOT stuff is shared by multiple boards targeting different Zephyr versions 🥲

And meanwhile `soc.h` is still being included in our samples:

```
./zephyr/samples/drivers/kscan/src/main.c:10
./zephyr/samples/drivers/ps2/src/main.c:10
./zephyr/samples/drivers/led_xec/src/main.c:8
./zephyr/samples/drivers/clock_control_xec/src/main.c:8
./zephyr/samples/drivers/clock_control_xec/src/main.c:18
./zephyr/samples/drivers/espi/src/espi_oob_handler.c:8
./zephyr/samples/drivers/espi/src/main.c:10
./zephyr/samples/drivers/peci/src/main.c:10
./zephyr/samples/bluetooth/mesh_demo/src/microbit.c:10
./zephyr/samples/boards/mec172xevb_assy6906/rom_api/src/main.c:17
./zephyr/samples/boards/mec172xevb_assy6906/qmspi_ldma/src/main.c:10
./zephyr/samples/boards/esp32/flash_memory_mapped/src/main.c:14
./zephyr/samples/boards/esp32/flash_encryption/src/main.c:15
./zephyr/samples/boards/mec15xxevb_assy6853/power_management/src/power_mgmt.c:10
```

and a lot of other places of course..

**Pull request merged**